### PR TITLE
Add docs for i18n and unify debug-locales to run per app

### LINKED ISSUE #417: Document the localisation processes involved in the addons-frontend project
Should cover setting up new locales and the extraction and merging procedures.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #417

* Unifies all the locale commands to require `NODE_APP_INSTANCE`
* Adds docs for the commands

### CODE REVIEW

File: bin/debug-locales [hunk 1]
```diff
@@ -12,27 +13,32 @@
...
 for (locale of debugLocales) {
-  for (potFile of potFiles) {
-    fileName = path.basename(potFile, '.pot');
-    poFile = path.join(localeDir, locale, 'LC_MESSAGES', `${fileName}.po`);
-    rewrite = locale === 'dbl' ? 'unicode' : 'flipped';
-    shell.exec(`podebug -i ${potFile} -o ${poFile} --rewrite ${rewrite}`);
-  }
+  potFile = path.join(templateDir, `${appName}.pot`);
+  poFile = path.join(localeDir, locale, 'LC_MESSAGES', `${appName}.po`);
+  rewrite = locale === 'dbl' ? 'unicode' : 'flipped';
```
Comment (mstriemer) [inline bin/debug-locales]: I know this is old but it's always bugged me. I'd rather see something like:

```js
const debugLocales = [
  ['dbl', 'unicode'],
  ['dbr', 'flipped'],
];
for ([locale, rewrite] of debugLocales) {
  // ...
}

File: docs/i18n.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+These docs detail the steps needing to be run to extract newly added strings
+and update the locales.
+
+After running each process you will need to commit the changes and push a PR
+to the addons-frontend repo.
+
+### Extracting newly added strings
+
+To extract new strings for an app in the addons-frontend project you should
```
Comment (mstriemer) [inline docs/i18n.md]: What does extracting new string do?

**Comment by mstriemer** [comment]
Thanks for documenting it! r+wc

File: bin/debug-locales (hunk 1 above)
Comment (muffinresearch) [inline bin/debug-locales]: Looks like there's not enough es6 capability to do this since these are just node scripts.

**Pull request merged**